Editor/Deployment/DeploymentSettings.cs

```diff
@@ -96,15 +106,15 @@ public int ScenarioIndex
             && IsBuildFilePathFilled;
 
         public bool IsBuildFilePathFilled => IsValidScenarioIndex
-            && (!_deployers[ScenarioIndex].HasGameServer || _coreApi.FileExists(BuildFilePath));
+            && (!_deployers[Scenario].HasGameServer || _coreApi.FileExists(BuildFilePath));
 
         public bool IsBuildFolderPathFilled => IsValidScenarioIndex
-           && (!_deployers[ScenarioIndex].HasGameServer || _coreApi.FolderExists(BuildFolderPath));
+           && (!_deployers[Scenario].HasGameServer || _coreApi.FolderExists(BuildFolderPath));
 
         public bool IsBuildRequired =>
-            IsValidScenarioIndex && _deployers[ScenarioIndex].HasGameServer;
+            IsValidScenarioIndex && _deployers[Scenario].HasGameServer;
 
-        public bool IsValidScenarioIndex => ScenarioIndex >= 0 && ScenarioIndex < _deployers.Count;
+        public bool IsValidScenarioIndex => Enum.IsDefined(typeof(DeploymentScenarios), _scenario) && _deployers.ContainsKey(_scenario);
```

Comment:
This should just be `IsValidScenario`now

Reply:
How exactly does a user come upon a InvalidScenario example? Is that even possible if we allow what they can select from UX. If this is not do-able we should remove the variable altogether to simplify code.

Reply:
We need to validate that the settings when loaded from file still matches an available scenario as it could be modified outwith the plugin